Import WPT css/css-nesting

mdubet opened the pull request:
#### ecaf5946d7a021d6e1d8bda4bb6e2c990a73d26a
<pre>
Import WPT css/css-nesting
<a href="https://bugs.webkit.org/show_bug.cgi?id=248608">https://bugs.webkit.org/show_bug.cgi?id=248608</a>

Reviewed by Antti Koivisto and Tim Nguyen.

WPT 67ccf32

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-properties-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-properties-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-properties.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-rules-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-rules-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-rules.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/cssom-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/cssom.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/implicit-nesting-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/implicit-nesting-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/implicit-nesting.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/implicit-parent-insertion-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalid-inner-rules-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalid-inner-rules.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-001-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-002-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-003-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-004-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-004.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/nesting-basic-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/nesting-basic-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/nesting-basic.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/parsing-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/parsing.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/pseudo-part-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/serialize-group-rules-with-decls.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/serialize-group-rules-with-decls.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/w3c-import.log: Added.

Canonical link: <a href="https://commits.webkit.org/257367@main">https://commits.webkit.org/257367@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b0610c7a50790515baac295d212b9507a7d61df2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98753 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7964 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31899 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108169 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168423 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8499 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85332 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91282 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106086 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104435 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/6437 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90009 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33419 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/88261 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/21351 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76344 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1881 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/22881 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1788 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45345 "Found 1 new test failure: media/video-inactive-playback.html (failure)") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/74/builds/5061 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6737 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42336 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2546 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3183 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Tools/Scripts/import-w3c-tests:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Was this intended? Can we undo this?

mdubet replied:
Not in this PR indeed. Removed

mdubet committed:
Import WPT css/css-nesting
https://bugs.webkit.org/show_bug.cgi?id=248608

Reviewed by Antti Koivisto and Tim Nguyen.

WPT 67ccf32

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-properties-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-properties-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-properties.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-rules-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-rules-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/conditional-rules.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/cssom-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/cssom.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/implicit-nesting-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/implicit-nesting-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/implicit-nesting.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/implicit-parent-insertion-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalid-inner-rules-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalid-inner-rules.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-001-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-002-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-003-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-004-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/invalidation-004.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/nesting-basic-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/nesting-basic-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/nesting-basic.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/parsing-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/parsing.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/pseudo-part-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/serialize-group-rules-with-decls.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/serialize-group-rules-with-decls.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-nesting/w3c-import.log: Added.

Canonical link: https://commits.webkit.org/257367@main
